independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java

```diff
@@ -221,10 +221,10 @@ protected boolean addDependency(ArtifactCoords coords, boolean managed) {
         } else if (model().getDependencies()
                 .stream()
                 .noneMatch(thisDep -> d.getManagementKey().equals(thisDep.getManagementKey()))) {
-            model().addDependency(d);
+            model().getDependencies().add(0, d);
```

Comment:
What was the reason for this?

Reply:
So the extension are added at the top of the dependency list. Most people have tests dependencies at the bottom, adding them under is a real pain.

Reply:
So the extension are added at the top of the dependency list. Most people have tests dependencies at the bottom, adding them under is a real pain.

Reply:
I am -1 on this one. It's not really solving anything. Keep in mind that the order of the dependencies is significant unless they are managed by a BOM. If you don't want to add extensions after the test dependencies, you can iterate through the dependencies and look for the first one with `test` scope.